tests/federation_room_join_partial_state_test.go

```diff
@@ -121,6 +121,44 @@ func (s *server) AddEDUHandler(eduHandler func(gomatrixserverlib.EDU) bool) func
 	}
 }
 
+// WithWaitForLeave runs the given action and waits for the user to leave the room.
+func (s *server) WithWaitForLeave(
+	t *testing.T, room *federation.ServerRoom, userID string, leaveAction func(),
+) {
+	leaveChannel := make(chan *gomatrixserverlib.Event, 10)
+	removePDUHandler := s.AddPDUHandler(
+		func(e *gomatrixserverlib.Event) bool {
+			if membership, _ := e.Membership(); e.Type() == "m.room.member" &&
```

Comment:
what's the second thing returned by `Membership()` that gets bound(?) to `_` -- a  state key?

Reply:
It's `err` (surprise!). `membership` is `""` when there's no such state entry, so I elided the `err` check.